replicators/src/mysql_connector/connector.rs

```diff
@@ -128,8 +132,27 @@ impl MySqlBinlogConnector {
         };
 
         connector.register_as_replica().await?;
-        connector.request_binlog().await?;
-
+        let binlog_request = connector.request_binlog().await;
+        match binlog_request {
+            Ok(()) => (),
+            Err(mysql_async::Error::Server(ref err)) if err.code == 1236 => {
+                // Requested binlog is not available, we need to re-snapshot
+                return Err(ReadySetError::FullResnapshotNeeded);
+            }
+            Err(mysql_async::Error::Other(ref err)) => {
+                if let Some(rs_err) = err.downcast_ref::<ReadySetError>() {
+                    if rs_err == &ReadySetError::FullResnapshotNeeded {
```

Comment:
```suggestion
                if err.downcast_ref::<ReadySetError>() == Some(&ReadySetError::FullResnapshotNeeded) {
```

maybe... or use `matches!` if that doesn't work

Reply:
Done